gearcatalog/frontend/src/components/gear/Form.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from "react";
+import { connect } from "react-redux";
+import { addGear } from "../../actions/gear-list";
+import { Link } from "react-router-dom";
+import Input from '../common/Input';
+import { useInput } from '../../hooks/useInput';
+
+type FormProps = {
+  // TODO: find correct gear type
+  addGear: (gear) => void,
+};
+
+const Form = ({ addGear }: FormProps) => {
+  const [name, setName] = useState('');
+  const [desc, setDesc] = useState('');
+  const [brand, setBrand] = useState('');
+  const [weight_grams, setWeight] = useState('0');
+  const [length_mm, setLength] = useState('0');
+  const [width_mm, setWidth] = useState('0');
+  const [depth_mm, setDepth] = useState('0');
+  const { value:locking, bind:bindLocking, reset:resetLocking } = useInput('false');
+
+  // TODO: Find correct event type
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const gear = {
+      name,
+      desc,
+      brand,
+      weight_grams,
+      length_mm,
+      width_mm,
+      depth_mm,
+      locking,
+    };
+    addGear(gear);
+    setName('');
+    setDesc('');
+    setBrand('');
+    setWeight('0');
+    setLength('0');
+    setWidth('0');
+    setDepth('0');
+    resetLocking();
+  };
+
+  return (
+    <div className="card card-body mt-2 mb-2 pb-1 shadow">
+      <h2>Add Gear</h2>
+      <form onSubmit={handleSubmit}>
+        <Input
+          name={'Name'}
+          type={'text'}
+          val={name}
+          setVal={setName}
+          />
```

Comment:
Why have all these parameters unwrapped? It creates a lot of wasted vertical space.

```tsx
(<Input name={'Name'} type={'text'} val={name} setVal={setName} />)
```

Reply:
JetClosing did it that was and I just got used to doing it that way. I am honestly unsure what is better and was trying it out that way for a bit.

Reply:
It is your codebase, so your choice, but my preference would be to only unwrap parameters like this when they are too long to fit on one line. This is the same as function calls in my mind.

Reply:
I think that is what I would prefer too. So I will change it.